src/Frontend/camFrontend/src/app/components/result-presentation/result-presentation.component.ts

```diff
@@ -43,11 +43,19 @@ export class ResultPresentationComponent {
   saveResult(result: Result, finalAspectList: Array<string>) {
     this.finalAspectList = finalAspectList;
 
-    // count the number of sentences used for comparison
+    // counts the number of sentences used for comparison
     this.sentenceCount = result.sentenceCount;
 
-    this.saveWinner(result.object1.name, result.object2.name);
-    this.saveScores(result.object1.points, result.object2.points, result.object1.totalPoints, result.object2.totalPoints);
+    // allocates the winner or looser object. The first parameter will be the winner and second will be looser 
+    if (result.winner === result.object1.name){
+      this.saveWinner(result.object1.name, result.object2.name);
+      this.saveScores(result.object1.points, result.object2.points, result.object1.totalPoints, result.object2.totalPoints);
+    }else{
+      this.saveWinner(result.object2.name, result.object1.name);
+      this.saveScores(result.object2.points, result.object1.points, result.object2.totalPoints, result.object1.totalPoints);
```

Comment:
Yes, it will always go first. This snippet of code has nothing to do with that. I just kept the information regarding winner and loser and their respective scores although these variables aren't being utilized, I left them as it is in case of need in the future!

Reply:
It is running perfectly fine on my side. It would be nice if you could pull and run the code at the development branch.